dsgrid/registry/project_registry_manager.py

```diff
@@ -1074,74 +1063,67 @@ def _check_dataset_base_to_project_base_mappings(
-            for col in diff.columns:
-                diff_counts[col] = diff.select(col).distinct().count()
-            diff.unpersist()
-            dim_table.cache()
-            dataset_counts = {}
-            for col in diff.columns:
-                dataset_counts[col] = dim_table.select(col).distinct().count()
-                if dataset_counts[col] != diff_counts[col]:
-                    logger.error(
-                        "Error contributor: column=%s dataset_distinct_count=%s missing_distinct_count=%s",
-                        col,
-                        dataset_counts[col],
-                        diff_counts[col],
-                    )
-            dim_table.unpersist()
-
-            raise DSGInvalidDataset(
-                f"Dataset {dataset_config.config_id} is missing required dimension records. "
-                "Please look in the log file for more information."
-            )
-        if pivoted_dimension is not None:
-            project_pivoted_ids = project_config.get_required_dimension_record_ids(
-                dataset_id, pivoted_dimension
-            )
-            self._check_pivoted_dimension_columns(handler, project_pivoted_ids, dataset_id)
+        raise DSGInvalidDataset(
+            f"Dataset {dataset_config.config_id} is missing required dimension records. "
+            "Please look in the log file for more information."
+        )
 
-    @staticmethod
-    def _check_pivoted_dimension_columns(handler, project_pivoted_ids, dataset_id):
-        """pivoted dimension record is the same as project's unless a relevant association is provided."""
-        logger.info("Check pivoted dimension columns.")
-        d_dim_ids = handler.get_pivoted_dimension_columns_mapped_to_project()
-        pivoted_dim = handler.config.get_pivoted_dimension_type()
-
-        if d_dim_ids.symmetric_difference(project_pivoted_ids):
-            raise DSGInvalidDataset(
-                f"Mismatch between project and {dataset_id} dataset pivoted {pivoted_dim.value} dimension, "
-                "please double-check data, and any relevant association_table and dimension_mapping. "
-                f"\n - Invalid column(s) in {dataset_id} load data according to project: {d_dim_ids.difference(project_pivoted_ids)}"
-                f"\n - Missing column(s) in {dataset_id} load data according to project: {project_pivoted_ids.difference(d_dim_ids)}"
-            )
+    @track_timing(timer_stats_collector)
+    def _make_dimension_associations(
+        self,
+        config: ProjectConfig,
+        dataset_id: str,
+        context: ScratchDirContext,
+    ):
+        logger.info("Make dimension association table for %s", dataset_id)
+        path = context.get_temp_filename()
+        df = config.make_dimension_association_table(dataset_id, context)
+        df = write_dataframe_and_auto_partition(df, path)
+        context.add_tracked_path(path)
+        logger.info("Wrote dimension associations for dataset %s", dataset_id)
```

Comment:
nit: Tell the user where the dimension associations were written to?

Reply:
I would say no. We are going to delete it when finished. Do we need an option to keep these around? If that is needed then perhaps we need specific CLI commands to create project and dataset dimension association files.